### PR TITLE
Tweak: UI & Utilization for Post Excerpt Generation- Elementor Editor [ED-15189]

### PR DESCRIPTION
**Pull request by naorsabag**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* implement AI excerpt on the editor side

## Description
An explanation of what is done in this PR

image
image
image

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/ai/assets/js/editor/ai-behavior.js [hunk 1]
```diff
@@ -28,6 +30,37 @@ export default class AiBehavior extends Marionette.Behavior {
 		};
 	}
 
+	getTextualContent() {
```
Comment (naorsabag) [inline modules/ai/assets/js/editor/ai-behavior.js:33]: not sure about this method location, any idea?

File: modules/ai/assets/js/editor/ai-behavior.js [hunk 2]
```diff
@@ -28,6 +30,37 @@ export default class AiBehavior extends Marionette.Behavior {
...
+		const pageTextContent = [];
+
+		// Use elementorFrontend to loop through all the elements in the Elementor editor
+		const clonedCurrentLayout = window.elementor.$previewContents[ 0 ]?.cloneNode( true );
+		clonedCurrentLayout.querySelectorAll( '.elementor-editor-element-settings, #elementor-add-new-section, .elementor-add-section-inner, header, footer' )
```
Comment (matipojo) [inline modules/ai/assets/js/editor/ai-behavior.js:38]: The content may include header and footer tags

File: modules/ai/assets/js/editor/ai-behavior.js [hunk 3]
```diff
@@ -40,21 +73,32 @@ export default class AiBehavior extends Marionette.Behavior {
...
-				additionalOptions={ this.getOption( 'additionalOptions' ) }
-				onClose={ () => {
-					unmount();
-					rootElement.remove();
-				} }
-				colorScheme={ colorScheme }
-				isRTL={ isRTL }
-			/>
-		), rootElement );
+		const { unmount } = 'excerpt' === this.getOption( 'type' )
```
Comment (matipojo) [inline modules/ai/assets/js/editor/ai-behavior.js:76]: Using an `if` will make it more readable

**Review by matipojo — DISMISSED** [review]

File: modules/ai/assets/js/editor/module.js [hunk 4]
```diff
@@ -135,6 +135,27 @@ export default class Module extends elementorModules.editor.utils.Module {
 			}
 		}
 
+		if ( 'excerpt' === aiOptions.type ) {
+			behaviors.ai = {
+				behaviorClass: AiBehavior,
+				type: aiOptions.type,
+				getControlValue: view.getControlValue.bind( view ),
+				setControlValue: ( value ) => {
+					if ( 'wysiwyg' === controlType ) {
```
Comment (netanelavr) [inline modules/ai/assets/js/editor/module.js:144]: ?

File: modules/ai/assets/js/editor/ai-behavior.js [hunk 5]
```diff
@@ -28,6 +30,37 @@ export default class AiBehavior extends Marionette.Behavior {
...
+				if ( cleanText ) {
+					pageTextContent.push( cleanText );
+				}
+			} else {
+				// If not a text node, iterate over its child nodes
+				node.childNodes.forEach( ( child ) => walkDOM( child, ancestorOfElementorElement ) );
+			}
+		};
+
+		walkDOM( clonedCurrentLayout );
```
Comment (netanelavr) [inline modules/ai/assets/js/editor/ai-behavior.js:58]: consider extracting it to a separate method

**Review by netanelavr — DISMISSED** [review]

**Comment by netanelavr** [comment]
@naorsabag please add some visuals to the PR description

File: modules/ai/assets/js/editor/ai-behavior.js (hunk 2 above)
Comment (naorsabag) [reply modules/ai/assets/js/editor/ai-behavior.js:38]: I added header and footer to the query, why wouldn't it be removed?

File: modules/ai/assets/js/editor/module.js (hunk 4 above)
Comment (naorsabag) [reply modules/ai/assets/js/editor/module.js:144]: this is how we update textual component with our generated text. see https://github.com/elementor/elementor/blob/9500508aa2579523674a60d9fa19f2b9711d7430/modules/ai/assets/js/editor/module.js#L78C36-L79C1

**Review by netanelavr — APPROVED** [review]

**Pull request merged**